react/daily-art-prompt/src/store/rootReducer.test.js

```diff
@@ -1,573 +1,506 @@
+          ];
+          const action = {
+            type: TYPES.EDIT_COMMENT,
+            payload: {
+              imageId: 1,
+              commentId: 2
             }
-            expect(promptsReducer(state, action)).toEqual(state)
-        })
-
-    })
+          };
+          const result = promptImagesReducer(state, action);
+          expect(result[0].comments[0]).toEqual(state[0].comments[0]);
+        });
+      });
+    });
+    describe("when image.id does not equal imageId given", () => {
+      it("should return image unchanged", () => {
+        const state = [
+          {
+            id: 1,
+            comments: [
+              {
+                id: 1
+              }
+            ]
+          }
+        ];
+        const action = {
+          type: TYPES.EDIT_COMMENT,
+          payload: {
+            imageId: 2,
+            commentId: 1
+          }
+        };
+        const result = promptImagesReducer(state, action);
+        expect(result[0]).toEqual(state[0]);
+      });
+    });
+  });
 
+  describe("when action.type does not equal any of the cases", () => {
+    describe("when goes to the default case", () => {
+      it("should return the state unchanged", () => {
+        const state = [];
+        const action = {
+          type: ""
+        };
+        expect(promptImagesReducer(state, action)).toEqual(state);
+      });
+    });
+  });
+});
 
-})
+describe("dateReducer", () => {
+  describe("when action.type is SET_INITIAL_DATE", () => {
+    it("should return correct date with correct format", () => {
+      const action = {
+        type: TYPES.SET_INITIAL_DATE
+      };
+      expect(dateReducer(null, action)).toEqual(moment().format("YYYY-MM-DD"));
+    });
+  });
+  describe("when action.type is UPDATE_NEXT_DATE", () => {
+    it("should return the date from action payload", () => {
+      const action = {
+        type: TYPES.UPDATE_NEXT_DATE,
+        payload: {
+          date: "2020-01-12"
+        }
+      };
+      expect(dateReducer(null, action)).toEqual("2020-01-12");
+    });
+  });
+  describe("when action.type is UPDATE_PREVIOUS_DATE", () => {
+    it("should return the date from action.payload", () => {
+      const action = {
+        type: TYPES.UPDATE_PREVIOUS_DATE,
+        payload: {
+          date: "2020-01-10"
+        }
+      };
+      expect(dateReducer(null, action)).toEqual("2020-01-10");
+    });
+  });
+  describe("when the case type defaults", () => {
+    it("should return state", () => {
+      const state = "2020-01-11";
+      const action = {
+        type: ""
+      };
+      expect(dateReducer(state, action)).toEqual(state);
+    });
+  });
+});
 
+describe("promptsReducer", () => {
+  describe("when action.type is SET_INITIAL_PROMPTS", () => {
+    it("should return initial prompts", () => {
+      const state = [];
+      const action = {
+        type: TYPES.SET_INITIAL_PROMPTS,
+        payload: {
+          prompts: ["array of prompts"]
```

Comment:
this doesn't care what prompts are (and technically they are not arrays)

Reply:
well technically!!!! ... fine i changed it